assets/src/edit-story/components/canvas/canvasProvider.js

```diff
@@ -3,43 +3,43 @@
  */
 import PropTypes from 'prop-types';
 
-/**
- * WordPress dependencies
- */
-import { useState, useCallback } from '@wordpress/element';
-
 /**
  * Internal dependencies
  */
+import useFunctionState from '../../utils/useFunctionState';
+import useEditingElement from './useEditingElement';
 import Context from './context';
 
 function CanvasProvider( { children } ) {
-	const [ isEditMode, setIsEditMode ] = useState( false );
-	// It's a bit weird to directly set a state to be a function (as setFoo calls
-	// any function given to unwrap the inner value, which can then be a function),
-	// so we use a wrapper object in stead of double-functioning.
-	// We create helper functions below to set it directly.
-	const [ backgroundClick, setBackgroundClick ] = useState( { handler: null } );
-
-	const setBackgroundClickHandler = useCallback(
-		( handler ) => setBackgroundClick( { handler: typeof handler === 'function' ? handler : null } ),
-		[ setBackgroundClick ],
-	);
-
-	const clearBackgroundClickHandler = useCallback(
-		() => setBackgroundClick( { handler: null } ),
-		[ setBackgroundClick ],
-	);
+	const {
+		editingElement,
+		editingElementState,
+		setEditingElementWithState,
+		setEditingElementWithoutState,
+		clearEditing,
+		setNodeForElement,
+	} = useEditingElement();
+
+	const [
+		backgroundClickHandler,
+		setBackgroundClickHandler,
+		clearBackgroundClickHandler,
```

Comment:
I originally thought it would be used for multiple purposes, (like exiting edit mode), but it turns out to not be the case. I suggest keeping it for now, so we have the option of changing what happens when you click on the background under special circumstances, but it's a good idea to mark it as potentially obsolete.